### ISSUE TITLE
Autodrive not picking up to refill or unload from CP course

### ISSUE THREAD

**Issue by authenticx** [Stephan-S/FS25_AutoDrive#65]
When a worker needs to refill lime/fertilizer/seeds etc or when my forage worker needs to go unload the wagon, CP is not handing off to autodrive for the wrorker to go refill or unload then return to resume the CP course.

**Comment by Iwan1803** [comment]
Did you activate the CP button in the AD HUD?

**Comment by Zofaqxx** [comment]
Yep, das kann ich bestätigen. Bei mir ging das nur so halb, da ich mit dem Ladewagen Gras gesammelt habe. Er hat es zum Silo gebracht, hat aber nicht wieder angefanden, musste dann AD ausschalten und bei CP wieder starten. Beim Düngen füllt er nicht wieder auf, wenn er leer ist. Also zusammen gefasst, AD übergibt nicht an CP. Hat zumindest bei mir, bei keiner Aufgabe funktioniert.

**Comment by prawler74** [comment]
Hi

Same issue here. CP does not hand over to AD.

**Comment by Zofaqxx** [comment]
> Hi
> 
> Same issue here. CP does not hand over to AD.

You have to right clicked, then you have the CP Button.

**Comment by Zofaqxx** [comment]
> Right click?
> 
> Not sure i understand...i click the wheel in Autodtive.
> 
> The CP button was used to do this in FS22.
> 
> Am i missing something?😊
> 
> Bjørn

You have to right click on the steering wheel once and then the CP button will appear and then you can left click to activate it.

**Comment by prawler74** [comment]
Hi

Thanx for explaining, will give it a try

Bjørn

**Comment by prawler74** [comment]
Now comes the next issue, have you had any success with unloading from a combine? It misses the pipe, and yes i have tried to change thesettings, but it still will not get the trailer/tipper to get under it to unload. Have to do that my self......

**Comment by Zofaqxx** [comment]
> Now comes the next issue, have you had any success with unloading from a combine? It misses the pipe, and yes i have tried to change thesettings, but it still will not get the trailer/tipper to get under it to unload. Have to do that my self......

No, i doesn‘t have this problem, for me the combine unload to the trailer.

**Comment by Zofaqxx** [comment]
With combine you use? Only the case 7xxx combine have a problem with unloading, but this problem is from Giants.

**Comment by prawler74** [comment]
Hi...yes i am using the Case 7150. I got it to unload when setting the distance to -0,1.....but that should not be an issue....Will be changing that one out...thanx for the tip

**Comment by Zofaqxx** [comment]
No Problem
 I saw this issue by CP, there is someone who posted it.

**Comment by authenticx** [comment]
Yes I right click the steering wheel and then left clicp the CP button and I then start the courseplay course.  I have done what I have done in past versions when using CP and AD which is go to the start of the CP course and record a "field exit" with autodrive and the path to the refill station and back to the field.  I set the waypoint for refill past the trigger so the worker will drive a bitg past to account for the lime spreader.  What happens for example is, if my lime worker runs out during the course he just stops.  At that point if I try to start the autodrive in any form it jhust says it can't get to "Lime  Refill" or can't calculate path.  If I drive to the start of the AD path and start AD then start AD it drives to the point, refills the spreader and drives back so that part is ok.  

I seem to remember in the past having to "link" the CP course and the AD path by joining the markers?  Maybe I am not remembering correctly but in any case I can't find a way to do that here or maybe it isn't supposed to be neccesary anymore.  

I realize it is quite possible maybe I am simply not doing something correctly so if that is the case i apologize for wasting time.

Here is what I try to do and I will use my forage worker picking up grass for example:

 I setup like this with Mowing course since I would be following the swath dropped by the Big M 450 mower. The Unload Hay waypoint is set as my deliver. CP is chosen and turned on.  The red line is my autodrive path I recorded with the waypoint at the silo trigger to unload the grass.  When the forager is full, the worker just stops and does not drive to the start of the autodrive route to go unload.

![image](https://github.com/user-attachments/assets/b0507d61-3455-49fe-bb51-c898d4112210)

**Comment by Draku1986** [comment]
Du brauchst den Modus Abholen und abliefern, Im Oberen Feld die Position am Feld angeben, und im Zweiten den Entladepunkt.
Das dritte wo man das Transportgut auswählen kann, kann auch bleiben wie in meinem Screenshot. Natürlich noch Courseplay-Butten aktivieren, und los gehts
![Screenshot 2025-01-01 120028](https://github.com/user-attachments/assets/e9344062-0605-40b9-bf90-6a1d55dfcd98)

**Comment by erksartori** [comment]
About the CP button, it is not saved, every time I save the game I go to the next day or even the same day at another time, I have to keep clicking on it, would there be a way to define a usage pattern in the tractor settings, as this can be different for each type of equipment and tractor.

**Comment by authenticx** [comment]
Ok after now understanding the postion waypoints and adjusting some routes I have got this to work.  The only thing now I wonder is why do the workers drive sooooo sloooow on hte recorded routes?  They seem to ignore the speed limits set in the autodrive window. It is taking forever for workers to drive to sell or refill etc..

**Comment by Iwan1803** [comment]
@authenticx >80% of driving "too slow" is due to the route network and especially at curves and junctions.
The angles between the waypoints are decisive for the speed calculated by AD.
less waypoints - bigger angels - slower speed
more waypoints - smaller angels - higher speed

**Comment by ThomasMatern** [comment]
Please join the discord server for help. I'm closing this ticket since we're no longer discussing the actual "issue" reported.
